src/commands/stepfunctions/helpers.ts

```diff
@@ -237,13 +246,43 @@ check out https://docs.datadoghq.com/serverless/step_functions/troubleshooting/\
     return true
   }
 
-  // payload is not a JSON object
-  if (step.Parameters.hasOwnProperty('Payload') && typeof step.Parameters['Payload'] !== 'object') {
-    context.stdout.write(`[Warn] Step ${stepName}'s Payload field is not a JSON object. Step Functions Context Object \
+  if (step.Parameters.hasOwnProperty('Payload')) {
+    if (typeof step.Parameters['Payload'] !== 'object') {
+      // payload is not a JSON object
+      context.stdout
+        .write(`[Warn] Step ${stepName}'s Payload field is not a JSON object. Step Functions Context Object \
 injection skipped. Your Step Functions trace will not be merged with downstream Lambda traces. To manually \
 merge these traces, check out https://docs.datadoghq.com/serverless/step_functions/troubleshooting/\n`)
 
-    return false
+      return false
+    } else {
+      const payload = step.Parameters.Payload
+      if (payload['Execution.$'] === '$$.Execution' && payload['State.$'] === '$$.State') {
```

Comment:
Can we also check for `StateMachine` in this statement as the current layers require it to be set

Reply:
Good catch!